test/system/src/test/java/io/pravega/test/system/ReadWriteAndScaleWithFailoverTest.java

```diff
@@ -0,0 +1,488 @@
+             ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURIDirect)) {
+
+            log.info("Client factory details {}", clientFactory.toString());
+            //create writers
+            log.info("Creating {} writers", INITIAL_NUM_WRITERS);
+            List<EventStreamWriter<Long>> writerList = new ArrayList<>(INITIAL_NUM_WRITERS);
+            log.info("Writers writing in the scope {}", scope);
+            for (int i = 0; i < INITIAL_NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer = clientFactory.createEventWriter(STREAM_NAME,
+                        new JavaSerializer<Long>(),
+                        EventWriterConfig.builder().retryAttempts(10).build());
+                writerList.add(writer);
+            }
+
+            //create a reader group
+            log.info("Creating Reader group: {}, with readergroup manager using scope: {}", readerGroupName, scope);
+            readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                    Collections.singleton(STREAM_NAME));
+            log.info("Reader group name: {}, Reader group scope: {}, Online readers: {}",
+                    readerGroupManager.getReaderGroup(readerGroupName).getGroupName(), readerGroupManager
+                            .getReaderGroup(readerGroupName).getScope(), readerGroupManager
+                            .getReaderGroup(readerGroupName).getOnlineReaders());
+
+            //create readers
+            log.info("Creating {} readers", NUM_READERS);
+            List<EventStreamReader<Long>> readerList = new ArrayList<>(NUM_READERS);
+            log.info("Scope that is seen by readers {}", scope);
+            //start reading events
+            for (int i = 0; i < NUM_READERS; i++) {
+                log.info("Starting reader: {}, with id: {}", i, readerName + i);
+                final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                        readerGroupName,
+                        new JavaSerializer<Long>(),
+                        ReaderConfig.builder().build());
+                readerList.add(reader);
+            }
+
+            // start writing asynchronously
+            List<CompletableFuture<Void>> writerFutureList = writerList.stream().map(writer ->
+                    startWriting(eventData, writer, stopWriteFlag, eventWriteCount)).collect(Collectors.toList());
+
+            //start reading asynchronously
+            List<CompletableFuture<Void>> readerFutureList = readerList.stream().map(reader ->
+                    startReading(eventsReadFromPravega, eventWriteCount, eventReadCount, stopReadFlag, reader))
+                    .collect(Collectors.toList());
+
+            //run the failover test before triggering scaling
+            performFailoverTest();
```

Comment:
This is already tested in a different system test . We could remove this thereby reducing the system test execution time, right?